integration-tests/jest-reporter.js

```diff
@@ -37,16 +26,48 @@ module.exports = class JestReporter {
     // Write pass/failure summaries
     writableStream.write('\n\nTests Summary\n')
     _.forEach(result => {
+      const { title, status, failureMessages, failureDetails } = result
       writableStream.write('----------------------------------------------\n')
-      writableStream.write(`Test: ${result.title}\n`)
-      writableStream.write(`Status: ${result.status}\n`)
-      const { failureMessages = [] } = result
-      !_.isEmpty(failureMessages) && writableStream.write(`Failure message: ${failureMessages}\n`)
+      writableStream.write(`Test: ${title}\n`)
+      writableStream.write(`Status: ${status}\n`)
+      if (!_.isEmpty(failureMessages)) {
+        writableStream.write(`Failure messages: ${failureMessages}\n`)
+      }
+      if (!_.isEmpty(failureDetails)) {
+        writableStream.write(`Failure details: ${failureDetails}\n`)
+      }
       writableStream.write('\n')
     }, testResults)
 
     writableStream.end()
-    writableStream.on('finish', () => console.log(`Finished writing logs to ${logFileName}`))
+    writableStream.on('finish', () => console.log(`**  Saved ${name} results: ${logFileName}`))
+  }
+
+  // Called after all tests have completed
+  onRunComplete(_test, runResults) {
+    // Save run summary to a file.
+    const logDir = this.getLogPath(runResults)
+    const summaryFile = `tests-summary.json`
+    writeFileSync(`${logDir}/${summaryFile}`, JSON.stringify(runResults, null, 2))
+    console.log(`**  Saved all tests summary: ${logDir}/${summaryFile}`)
+    return runResults
+  }
+
+  getLogPath(result) {
+    const { testResults } = result
```

Comment:
In `onTestResult`, this function is passed an array of AssertionResults. That array would not have a `testResults` property, so this function would always return the `successes` directory for logs.